patches/ios-web_view-internal-cwv_web_view_configuration.mm.patch

```diff
@@ -0,0 +1,120 @@
+-  std::unique_ptr<ios_web_view::WebViewBrowserState> _browserState;
++  ios_web_view::WebViewBrowserState* _browserState;
+ 
+   // Holds all CWVWebViews created with this class. Weak references.
+   NSHashTable* _webViews;
+@@ -76,10 +77,13 @@ NSHashTable<CWVWebViewConfiguration*>* gNonPersistentConfigurations = nil;
+ + (instancetype)defaultConfiguration {
+   static dispatch_once_t onceToken;
+   dispatch_once(&onceToken, ^{
+-    auto browserState = std::make_unique<ios_web_view::WebViewBrowserState>(
+-        /* off_the_record = */ false);
++    ios::ChromeBrowserStateManager* browserStateManager =
++        GetApplicationContext()->GetChromeBrowserStateManager();
++    ChromeBrowserState* browserState =
++        browserStateManager->GetLastUsedBrowserStateDeprecatedDoNotUse();
+     gDefaultConfiguration = [[CWVWebViewConfiguration alloc]
+-        initWithBrowserState:std::move(browserState)];
++        initWithBrowserState:ios_web_view::WebViewBrowserState::
++                                 FromBrowserState(browserState)];
+   });
+   return gDefaultConfiguration;
+ }
+@@ -93,12 +97,15 @@ NSHashTable<CWVWebViewConfiguration*>* gNonPersistentConfigurations = nil;
+ }
+ 
+ + (CWVWebViewConfiguration*)nonPersistentConfiguration {
+-  CWVWebViewConfiguration* defaultConfiguration = [self defaultConfiguration];
+-  auto browserState = std::make_unique<ios_web_view::WebViewBrowserState>(
+-      /* off_the_record = */ true, defaultConfiguration.browserState);
++  ios::ChromeBrowserStateManager* browserStateManager =
++      GetApplicationContext()->GetChromeBrowserStateManager();
++  ChromeBrowserState* browserState =
++      browserStateManager->GetLastUsedBrowserStateDeprecatedDoNotUse()
++          ->GetOffTheRecordChromeBrowserState();
+   CWVWebViewConfiguration* nonPersistentConfiguration =
+       [[CWVWebViewConfiguration alloc]
+-          initWithBrowserState:std::move(browserState)];
++          initWithBrowserState:ios_web_view::WebViewBrowserState::
++                                   FromBrowserState(browserState)];
+ 
+   // Save a weak pointer to nonpersistent configurations so they may be shut
+   // down later.
+@@ -112,10 +119,10 @@ NSHashTable<CWVWebViewConfiguration*>* gNonPersistentConfigurations = nil;
+ }
+ 
+ - (instancetype)initWithBrowserState:
+-    (std::unique_ptr<ios_web_view::WebViewBrowserState>)browserState {
++    (ios_web_view::WebViewBrowserState*)browserState {
+   self = [super init];
+   if (self) {
+-    _browserState = std::move(browserState);
++    _browserState = browserState;
+ 
+     _preferences =
+         [[CWVPreferences alloc] initWithPrefService:_browserState->GetPrefs()];
+@@ -179,14 +186,14 @@ NSHashTable<CWVWebViewConfiguration*>* gNonPersistentConfigurations = nil;
+ #pragma mark - ReuseCheckService
+ 
+ - (CWVReuseCheckService*)reuseCheckService {
+-  if (!_reuseCheckService && self.persistent) {
+-    affiliations::AffiliationService* affiliation_service =
```

Comment:
this AffiliationService seems concerning in general, I asked @ShivanKaul and @pes10k about it because it looks like something we might want to just replace with a no-op implemenation

Reply:
Right now I don't compile any of the actual affiliation services (I just dont include them in the `//brave/ios/web_view` target) but likely we'd go the route of compiling `//ios/web_view` as is and no-op them with chromium_src overrides

Reply:
it looks like this is something to do with credential sharing with apps, but we already disable it by disabling the fetch

Reply:
was there a specific reason this was commented out?